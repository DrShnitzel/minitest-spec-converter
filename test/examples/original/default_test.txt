class ExampleTest < Minitest::Test
  def setup
    do_some_setup
  end

  def setup_some_not_minitest_setup
    do_something
  end

  def test_complex_case
    assert_equal "DTP*573*D8*#{date.strftime("%Y%m%d")}~", batch.get_tpl_denial_date(tpl_denial)
  end

  def test_truthy
    assert is_it_true?
  end

  def test_truthy_2
    assert(is_it_true?)
  end

  def test_some_value
    assert_equal some_value, expected_value
  end

  def test_some_value_2
    assert_equal(some_value, expected_value)
  end

  def test_not_equal
    assert_not_equal some_value, unexpected_value
  end

  def test_not_equal_2
    assert_not_equal(some_value, unexpected_value)
  end

  def test_nil
    assert_nil some_value
  end

  def test_nil_2
    assert_nil(some_value)
  end

  def test_not_nil
    refute_nil some_value
  end

  def test_not_nil_2
    refute_nil(some_value)
  end

  def test_empty
    assert_empty some_value
  end

  def test_empty_2
    assert_empty(some_value)
  end

  def test_not_empty
    refute_empty some_value
  end

  def test_not_empty_2
    refute_empty(some_value)
  end

  def test_includes
    assert_includes collection, item
  end

  def test_includes_2
    assert_includes(collection, item)
  end

  def test_not_includes
    refute_includes collection, item
  end

  def test_not_includes_2
    refute_includes(collection, item)
  end

  def test_match
    assert_match pattern, string
  end

  def test_match_2
    assert_match(pattern, string)
  end

  def test_not_match
    refute_match pattern, string
  end

  def test_not_match_2
    refute_match(pattern, string)
  end

  def test_raises
    assert_raises SomeException do
      method_that_raises
      line2
      line3
    end
  end

  def test_raises_2
    assert_raises(SomeException) { method_that_raises }
  end

  def test_instance_of
    assert_instance_of SomeClass, object
  end

  def test_instance_of_2
    assert_instance_of(SomeClass, object)
  end

  def test_not_instance_of
    refute_instance_of SomeClass, object
  end

  def test_not_instance_of_2
    refute_instance_of(SomeClass, object)
  end

  def test_kind_of
    assert_kind_of SomeClass, object
  end

  def test_kind_of_2
    assert_kind_of(SomeClass, object)
  end

  def test_not_kind_of
    refute_kind_of SomeClass, object
  end

  def test_not_kind_of_2
    refute_kind_of(SomeClass, object)
  end

  def test_predicate
    assert_predicate object, :predicate_method?
  end

  def test_predicate_2
    assert_predicate(object, :predicate_method?)
  end

  def test_not_predicate
    refute_predicate object, :predicate_method?
  end

  def test_not_predicate_2
    refute_predicate(object, :predicate_method?)
  end

  def test_operator
    assert_operator object, :>, other_object
  end

  def test_operator_2
    assert_operator(object, :>, other_object)
  end

  def test_not_operator
    refute_operator object, :>, other_object
  end

  def test_not_operator_2
    refute_operator(object, :>, other_object)
  end
end
