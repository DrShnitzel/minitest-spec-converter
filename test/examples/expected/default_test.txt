describe Example do
  before do
    do_some_setup
  end

  def setup_some_not_minitest_setup
    do_something
  end

  it "assert with multiple args" do
    _(ser2.eligible).must_equal(true)
  end

  it "hash vs named params" do
    _(named_arguments_are_not_hash(a: 1, b: 2)).must_equal({ a: 1, b: 2 })
  end

  it "complex case" do
    _(batch.get_tpl_denial_date(tpl_denial)).must_equal("DTP*573*D8*#{date.strftime("%Y%m%d")}~")
  end

  it "truthy" do
    _(is_it_true?).must_equal(true)
  end

  it "truthy 2" do
    _(is_it_true?).must_equal(true)
  end

  it "some value" do
    _(expected_value).must_equal(some_value)
  end

  it "some value 2" do
    _(expected_value).must_equal(some_value)
  end

  it "not equal" do
    _(unexpected_value).wont_equal(some_value)
  end

  it "not equal 2" do
    _(unexpected_value).wont_equal(some_value)
  end

  it "nil" do
    _(some_value).must_be_nil
  end

  it "nil 2" do
    _(some_value).must_be_nil
  end

  it "not nil" do
    _(some_value).wont_be_nil
  end

  it "not nil 2" do
    _(some_value).wont_be_nil
  end

  it "empty" do
    _(some_value).must_be_empty
  end

  it "empty 2" do
    _(some_value).must_be_empty
  end

  it "not empty" do
    _(some_value).wont_be_empty
  end

  it "not empty 2" do
    _(some_value).wont_be_empty
  end

  it "includes" do
    _(collection).must_include(item)
  end

  it "includes 2" do
    _(collection).must_include(item)
  end

  it "not includes" do
    _(collection).wont_include(item)
  end

  it "not includes 2" do
    _(collection).wont_include(item)
  end

  it "match" do
    _(string).must_match(pattern)
  end

  it "match 2" do
    _(string).must_match(pattern)
  end

  it "not match" do
    _(string).wont_match(pattern)
  end

  it "not match 2" do
    _(string).wont_match(pattern)
  end

  it "raises" do
    proc {
  method_that_raises
  line2
  line3
}.must_raise(SomeException)
  end

  it "raises 2" do
    proc {
  method_that_raises
}.must_raise(SomeException)
  end

  it "instance of" do
    _(object).must_be_instance_of(SomeClass)
  end

  it "instance of 2" do
    _(object).must_be_instance_of(SomeClass)
  end

  it "not instance of" do
    _(object).wont_be_instance_of(SomeClass)
  end

  it "not instance of 2" do
    _(object).wont_be_instance_of(SomeClass)
  end

  it "kind of" do
    _(object).must_be_kind_of(SomeClass)
  end

  it "kind of 2" do
    _(object).must_be_kind_of(SomeClass)
  end

  it "not kind of" do
    _(object).wont_be_kind_of(SomeClass)
  end

  it "not kind of 2" do
    _(object).wont_be_kind_of(SomeClass)
  end

  it "predicate" do
    _(object).must_be(:predicate_method?)
  end

  it "predicate 2" do
    _(object).must_be(:predicate_method?)
  end

  it "not predicate" do
    _(object).wont_be(:predicate_method?)
  end

  it "not predicate 2" do
    _(object).wont_be(:predicate_method?)
  end

  it "operator" do
    _(object).must_be(:>, other_object)
  end

  it "operator 2" do
    _(object).must_be(:>, other_object)
  end

  it "not operator" do
    _(object).wont_be(:>, other_object)
  end

  it "not operator 2" do
    _(object).wont_be(:>, other_object)
  end
end
